pkgs/by-name/ed/edopro/package.nix

```diff
@@ -0,0 +1,320 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchzip,
+  premake5,
+  writeShellScriptBin,
+  runCommandLocal,
+  symlinkJoin,
+  imagemagick,
+  bzip2,
+  curl,
+  flac,
+  # Use fmt 10+ after release 40.1.4+
+  fmt_9,
+  freetype,
+  libevent,
+  libgit2,
+  libGL,
+  libGLU,
+  libjpeg,
+  libpng,
+  libvorbis,
+  libX11,
+  libxkbcommon,
+  libXxf86vm,
+  lua5_3,
+  mono,
+  nlohmann_json,
+  openal,
+  SDL2,
+  sqlite,
+  wayland,
+  egl-wayland,
+  covers_url ? "https://pics.projectignis.org:2096/pics/cover/{}.jpg",
+  fields_url ? "https://pics.projectignis.org:2096/pics/field/{}.png",
+  # While ygoprodeck has higher quality images, "spamming" of their api results in a ban.
+  # Thats why this link can change since it's compiled into the program, However it will
+  # download assets when needed so it is unlikely to get banned. Unless you search the
+  # card list with no filters of any kind. When testing use ProjectIgnis' website instead.
+  pics_url ? "https://images.ygoprodeck.com/images/cards/{}.jpg",
+}:
+let
+  archLabel =
+    {
+      "x86_64-linux" = "x64";
+      "aarch64-linux" = "arm64";
+    }
+    .${stdenv.hostPlatform.system}
+      or (throw "Don't know how to translate ${stdenv.hostPlatform.system} into an arch label for edopro");
+
+  deps = import ./deps.nix;
+in
+let
+  assets = {
+    pname = "assets";
+    version = deps.edopro-version;
+
+    src = fetchzip {
+      url = "https://github.com/ProjectIgnis/edopro-assets/releases/download/${deps.edopro-version}/ProjectIgnis-EDOPro-${deps.edopro-version}-linux.tar.gz";
+      sha256 = deps.assets-hash;
+    };
+  };
+
+  irrlicht-edopro = stdenv.mkDerivation {
+    pname = "irrlicht-edopro";
+    version = "1.9-custom";
```

Comment:
`1.9-custom` doesn't really identify a version in any way that suits the `version` attribute. Would it be possible to have this be generated in our unstable version format `0-unstable-<YYYY>-<MM>-<DD>` as part of your update script?

Reply:
if you look at the project 1.9.0 is the version in the project.

Reply:
`1.9.0` doesn't name a specific rev either. The version is inherited from upstream Irrlicht, where it's been used as the "upcoming next minor release" for [years](https://github.com/edo9300/irrlicht1-8-4/commit/e09232760b3a1fa03d29a1d8f57c1597a0df6920#diff-a502ca260b30aa4885dd4fbe3e7a44bb1bdc0857e778580aba040c8705aaf924).

Reply:
Like is stated in the update script, irrlicht has no update structure. It's released as a rolling release, which is why I try to exit the update script before updating hashes.